packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -43,15 +43,13 @@ export class ApplicationExtension<
 
     /**
      * Called during server rendering and client application initialization. This method allows
-     * you to modify the routes of the base application, typically used to add new routes pointing
-     * at page components added by your application extension.
+     * you to add new routes, typically routes pointing at page components added by your application extension.
      *
      * @protected
-     * @param routes - The list application routes currently loaded.
-     * @returns routes - The modified application routes.
+     * @returns new routes to be added
      */
-    public extendRoutes(routes: RouteProps[]): RouteProps[] {
-        return routes
+    public getRoutes(params: GetRoutesParams): Promise<RouteProps[]> {
+        return Promise.resolve([])
```

Comment:
@vmarta after discussing with @bendvc  it seems we will go with having a `getRoutes` and `getRoutesAsync` method instead of `getRoutes` that can be either async/sync

In my code I attempted to implement the latter but the only way to check if `getRoutes` is async or not is by calling `getRoutes` and checking if a Promise is returned which Ben pointed out seemed wrong.

in my PR i have a `getRoutes` method that returns `RouteProps[]` and a `getRoutesAsync` that returns a `Promise<RouteProps[]>`. So not sure if we want to apply that in this PR or make that chagne as part of mine
https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2300/files#diff-eb67fe9398ed89ecd37d1e431a7599969fe7833eb831b0f0fe226c13b044eec9

Reply:
And I think after discussing with Ben it makes sense to make the default impl of `getRoutes` synchronous as that will be the common case

Reply:
@hajinsuha1 🤔 where should we implement the `getRoutes` and `getRoutesAsync`? I think I prefer in my PR.. but I'll refer to your code, especially the tests.

Reply:
That makes sense to me!